infra/examples-dev/gcp/main.tf

```
@@ -98,21 +98,26 @@ module "psoxy" {
   secret_replica_locations          = var.secret_replica_locations
   api_connectors                    = local.api_connectors
   bulk_connectors                   = local.bulk_connectors
-  webhook_collectors                = var.webhook_collectors
-  non_production_connectors         = var.non_production_connectors
-  custom_api_connector_rules        = var.custom_api_connector_rules
-  general_environment_variables     = var.general_environment_variables
-  pseudonymize_app_ids              = var.pseudonymize_app_ids
-  email_canonicalization            = var.email_canonicalization
-  bulk_input_expiration_days        = var.bulk_input_expiration_days
-  bulk_sanitized_expiration_days    = var.bulk_sanitized_expiration_days
-  custom_bulk_connector_rules       = var.custom_bulk_connector_rules
-  custom_bulk_connector_arguments   = var.custom_bulk_connector_arguments
-  lookup_tables                     = var.lookup_tables
-  custom_artifacts_bucket_name      = var.custom_artifacts_bucket_name
-  custom_side_outputs               = var.custom_side_outputs
-  todos_as_local_files              = var.todos_as_local_files
-  todo_step                         = local.max_auth_todo_step
+  webhook_collectors = { for k, v in var.webhook_collectors : k => merge(
+    v,
+    {
+      example_payload = file(v.example_payload_file)
+    }
+  ) }
```

example_payload_file is optional. Does "file" play nicely with nulls? Maybe you need a filter or a `coalesce(try(v.example_payload_file), null)`

```suggestion
  webhook_collectors = { for k, v in var.webhook_collectors if v.example_payload_file != null: k => merge(
    v,
    {
      example_payload = file(v.example_payload_file)
    }
  ) }
```

---

coalesce will fail if all null, but yeah, should be try with blank as fallback I think.